src/app/static/src/app/utils.ts

```diff
@@ -29,4 +30,19 @@ export const mapGettersByNames = <K extends string>(namespace: string, names: st
 export const mapActionsByNames = <K extends string>(namespace: string, names: string[]) => {
     type R = { [key in K]: any }
     return mapActions(namespace, names) as R
+
+};
+
+export const makeDownloadableContent = (data: string): string => {
+    const hash = CryptoJS.MD5(data);
+    return JSON.stringify([hash.toString(CryptoJS.enc.Base64), data]);
+};
+
+export const verifyDownloadableContent = (content: string): false | any => {
+    const result = JSON.parse(content);
+    const hash = result[0];
+    const data = result[1];
+    const valid = CryptoJS.MD5(data).toString(CryptoJS.enc.Base64) === hash;
+
+    return valid && JSON.parse(data);
```

Comment:
Made this so it returns false if invalid, otherwise returns the original object. Seem like the best behaviour?

Also I don't think either of these function names are great, better ideas?

Reply:
Agreed re return values. 

addChecksum / verifyChecksum ?